src/services/swap_service.ts

```diff
@@ -184,6 +185,32 @@ export class SwapService {
             signature: o.signature,
         }));
     }
+    private async _fetchTokenDecimalsIfRequiredAsync(tokenAddress: string): Promise<number> {
+        // HACK(dekz): Our ERC20Wrapper does not have decimals as it is optional
+        // so we must encode this ourselves
+        let decimals = findTokenDecimalsIfExists(tokenAddress, CHAIN_ID);
+        if (!decimals) {
+            const decimalsEncoder = new AbiEncoder.Method({
+                constant: true,
+                inputs: [],
+                name: 'decimals',
+                outputs: [{ name: '', type: 'uint256' }],
```

Comment:
just a small comment here, looks like this version of the erc20 abi defines the type of decimals as `uint8` https://github.com/ethereum/wiki/wiki/Contract-ERC20-ABI

Reply:
This shouldn't matter as everything gets padded to 32 bytes. But updated to be consistent.